ballerina/schema-client.bal

```diff
@@ -0,0 +1,320 @@
+// Copyright (c) 2022 WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/sql;
+
+public type TableDefinition record {
+    *sql:TableDefinition;
+    sql:CheckConstraint[] checkConstraints?;
+};
+
+# Represents an SQL metadata client.
+isolated client class SchemaClient {
+    private final Client dbClient;        
+    private final string database;
+
+    # Initializes the SchemaClient object
+    #
+    # + host - The name of where the postgresql server is hosted (ex: localhost)
+    # + user - The username to access the database
+    # + password - The password to access the database
+    # + database - The name of the database to be accessed
+    # + port - The port the database will be accessed on
+    # + options - PostgresSql database options
+    # + connectionPool - The `sql:ConnectionPool` to be used for the connection. If there is no
+    #                    `connectionPool` provided, the global connection pool (shared by all clients) will be used                  
+    # + return - An `sql:Error` or `()`
+    public function init(string host, string user, string password, string database, int port, 
+            Options? options = (), sql:ConnectionPool? connectionPool = ()) returns sql:Error? {        
+        self.database =  database;
+        self.dbClient = check new (host, user, password, database, port, options, connectionPool);
+    }
+
+    # Retrieves all tables in the database.
+    #
+    # + return - A string array containing the names of the tables or an `sql:Error`
+    isolated remote function listTables() returns string[]|sql:Error {
+        string[] tables = [];
+        stream<record {}, sql:Error?> results = self.dbClient->query(
+            `SELECT TABLE_NAME FROM INFORMATION_SCHEMA.TABLES 
+            WHERE TABLE_CATALOG = ${self.database} AND TABLE_SCHEMA NOT IN ('pg_catalog', 'information_schema');`
```

Comment:
Is it possible to have two similar table names if they exist in two different schema?

Reply:
It will retrieve all created tables from all schemas excluding any autogenerated tables (pg_catalog and information_schema). If there is a case where there are 2 tables with the same name from 2 different schemas then I believe it will retrieve both.